src/SFA.DAS.EmployerAccounts/Models/AccountTeam/Membership.cs

```diff
@@ -17,6 +17,7 @@ public virtual Role Role
             set => _role = value;
         }
 
+        // BUG: this is marked obsolete but the Role property relies on the RoleId property being set.
```

Comment:
okay, but then it is not obsolete "Marks the program elements that are no longer in use." - https://docs.microsoft.com/en-us/dotnet/api/system.obsoleteattribute?view=netframework-4.7.2  
The property *must* be set, and also set in the correct order so there is a temporal dependency in there as well.

Reply:
Ah I see now the issue should be that we set the role to None if role id is not set at all. Originally I thought you were saying the obsolete tag was the bug. The correct functionality should be that if the role and role Id are not set the a Role.None should be returned. This will then remove its dependancy. Given the time scale of this change we may even be able to remove this property now anyway.